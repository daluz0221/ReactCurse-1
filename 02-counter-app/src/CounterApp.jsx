import React, {useState} from 'react';
import PropTypes from 'prop-types';


const CounterApp = ({value}) => {


    const [counter, setCounter] = useState(value);

    
    

    //HandleAdd
    const HandleAdd = () => {
        //Ambas forman dan el mismoresultado
        // setCounter( counter + 1 );

        //esta es en caso de no poder acceder directamente al counter
        setCounter((c)=> c + 1);
    };

    const HandleMinus= () => {
        setCounter(counter - 1);
    };

    const HandleReset = () => {
        setCounter(value)
    };

    return (
        <>
         <h1>CounterApp</h1>
         <h2>{ counter }</h2>   

        <button
            onClick={HandleAdd}
        >+1</button>
        <button
            onClick={HandleReset}
        >Reset</button>
        <button
            onClick={HandleMinus}
        >-1</button>

        </>
    )
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

export default CounterApp;
