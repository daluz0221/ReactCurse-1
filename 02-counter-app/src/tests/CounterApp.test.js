import React from 'react'
import '@testing-library/jest-dom';

import {shallow} from 'enzyme'
import CounterApp from '../CounterApp';


describe('Pruebas en el componente <CounterApp />', () => {
    
    let wrapper = shallow(<CounterApp />);

    beforeEach( () =>  {
        wrapper = shallow(<CounterApp />);

    })
    
    test('debe de mostrar el <CounterApp /> correctamente', () => {
       
       expect( wrapper ).toMatchSnapshot();
    });
    test('debe de mostrar el valor enviado por props', () => {

        const value = 100;
        const wrapper = shallow(
        <CounterApp 
          value={value}            
        />);
        // con este find busco la etiqueta u objeto dentro del html del componente
        const textoParrafo = wrapper.find('h2').text();
        console.log(textoParrafo);

        expect(textoParrafo).toBe("100")

    });


    test('debe aumentar al click', () => {
        wrapper.find('button').at(0).simulate('click');
        const textoParrafo = wrapper.find('h2').text();

        expect(textoParrafo).toBe('13');

    });

    test('debe disminuir al click', () => {
        wrapper.find('button').at(2).simulate('click');
        const textoParrafo = wrapper.find('h2').text();

        expect(textoParrafo).toBe('11');

    });

    test('debe de colocar el valor por defectocon btn reset', () => {
        
        const wrapper = shallow( <CounterApp value={15} /> );
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const textoParrafo = wrapper.find('h2').text();

        expect(textoParrafo).toBe("15"); 

    });
    
    
    

});
