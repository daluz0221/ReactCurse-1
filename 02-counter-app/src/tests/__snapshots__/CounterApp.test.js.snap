// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <CounterApp /> debe de mostrar el <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    12
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
